Documentation/hacking.md

```diff
@@ -96,136 +96,49 @@ Follow the instructions on [Update coreos flavor stage1](devel/update-coreos-sta
-
+Use the glide tool to add a new dependency. In order to add a dependency to a package i.e. `github.com/fizz/buzz` for versions `>=1.2.3` and `<2.0.0`, execute:
 ```
-$ go get -d github.com/fizz/buzz
+$ glide get -u github.com/fizz/buzz#^1.2.3
+$ ./scripts/update-glide.sh
 ```
 
-##### If it is a library
+See the glide documentation about [Versions and Ranges](https://github.com/Masterminds/glide/blob/master/docs/versions.md).
 
-Add the new dependency into `godep`'s purview by simply importing the standard package name in one of your sources:
+*Note*: Do *not* use `go get` and `glide update` to add new dependencies. It will cause both `glide.lock` and `glide.yaml` files to diverge.
 
-```
-$ vim $GOPATH/src/github.com/coreos/rkt/some/file.go
-...
-import "github.com/fizz/buzz/tazz"
-...
-```
-
-Now, GOTO [saving](#saving-the-set-of-dependencies)
-
-##### If it is an application
-
-Add the new application to the `vendoredApps` file in the root of the repository:
-
-```
-$ vim vendoredApps
-...
-github.com/fizz/buzz/tazz
-...
-```
-
-Now, GOTO [saving](#saving-the-set-of-dependencies)
-
-#### Update an existing dependency
+#### Update existing dependencies
 
-The steps here are the same for both libraries and applications.
-In this case, assuming we're updating `github.com/foo/bar`:
+Once in a while new versions of dependencies are available. Entries in the `glide.yaml` file specify the target version ranges. If the version range does not suffice the target version, edit the `glade.yaml` directly, or leave it untouched if one wants to simply do a version bump of all dependencies.
 
+Execute the glide update script:
 ```
-$ cd $GOPATH/src/github.com/foo/bar
-$ git pull   # or 'go get -d -u github.com/foo/bar/...'
-$ git checkout $DESIRED_REVISION
-$ cd $GOPATH/src/github.com/coreos/rkt
-$ godep update github.com/foo/bar/...
+$ ./scripts/update-glide.sh
 ```
 
-Now, GOTO [saving](#saving-the-set-of-dependencies)
+*Note*: Changing specific entries in `glide.yaml` does not imply that only those will be updated. Glide will pull potential updates for all dependencies.
 
 #### Removing an existing dependency
 
-This is the simplest case of all.
-
-##### If it is a library
-
-Simply remove all references to a dependency from the source files.
-
-Now, GOTO [saving](#saving-the-set-of-dependencies)
-
-##### If it is an application
-
-Simply remove the relevant line from the `vendoredApps` file.
-
-Now, GOTO [saving](#saving-the-set-of-dependencies)
-
-### Saving the set of dependencies
-
-Finally, here we are, the magic command, the holy grail, the ultimate conclusion of all `godep` operations.
-Provided you have followed the preceding instructions, regardless of whether you are adding/removing/modifying dependencies, this innocuous script will cast the necessary spells to solve all of your dependency worries:
-
+Execute:
 ```
-$ ./scripts/godep-save
+$ glide rm github.com/fizz/buzz
+$ ./scripts/update-glide.sh
 ```
 
```

Comment:
> every glide-update.sh run will fetch all deps repos from scratch. That's quite a bit of network traffic, and should be probably mentioned.

Masterminds/glide#433 are some steps towards mitigating this, via a central repository cache. That should be in the next version of glide. vsolver takes the approach even further.